### ISSUE TITLE
Simulator fails in Windows venv.

### ISSUE THREAD

**Issue by tmarktaylor** [tmarktaylor/phmdoctest#16]
- The phmdoctest test suite has failures when run in a Python venv on Windows.
- When **simulator.run_and_pytest()** runs a subprocess to execute pytests the pytest module is not found.
- This occurs on Windows OS where pytest is only installed in the venv and not in the system (user) Python installation.
- The CI tests are not able to demonstrate the failure.

The problem is fixed by changing the **subprocess()** arg list: "python" is replaced with sys.executable.

**Comment by tmarktaylor** [comment]
This comment shows pytest failures caused by the bug.
It was reproduced in version v1.0.1 by running the tox cover environment in a windows machine.
tox installs phmdoctest in a Python venev.

### Todo
- [ ] Add tests that run phmdoctest in a Python venv on Windows.

#### Attached log file.
The attached GitHub action log file was created by the action source shown here:
Look for "No module named pytest" in the log file at the bottom.

```yml
name: Pytest/Coverage/Tox

on:
  workflow_dispatch:
  #push:
  #pull_request:

jobs:
  tests:
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [windows-latest]
        # os: [ubuntu-latest, windows-latest, macos-latest]
    steps:
    - uses: actions/checkout@v2
      # todo- remove before running in phmdoctest repos
      with:
        repository: tmarktaylor/phmdoctest
        ref: v1.0.1
    
    - name: Set up Python 3.x
      uses: actions/setup-python@v2
      with:
        python-version: 3.x
        
    - name: Install Tox
      run: |
        python -m pip install --upgrade pip
        pip install tox
    - name: Run tox cover
      run: tox -e cover
      #env:
      #  PYTHONPATH: ${{ github.workspace }}/src

```

[1_tests (windows-latest).txt](https://github.com/tmarktaylor/phmdoctest/files/6756929/1_tests.windows-latest.txt)

**Comment by tmarktaylor** [comment]
Fixed in version 1.2.1 uploaded to the Python Package Index.
